### PR TITLE
api/t/swarm: Fix deprecation for ServiceSpec.Networks

### PR DESCRIPTION
**Pull request by akerouanton**
**- What I did**

Fix the comment format to make linters able to catch the deprecation warning.

### CODE REVIEW

File: api/types/swarm/service.go [hunk 1]
```diff
@@ -34,9 +34,10 @@ type ServiceSpec struct {
 	UpdateConfig   *UpdateConfig `json:",omitempty"`
 	RollbackConfig *UpdateConfig `json:",omitempty"`
 
-	// Networks field in ServiceSpec is deprecated. The
-	// same field in TaskSpec should be used instead.
-	// This field will be removed in a future release.
+	// Networks specifies which networks the service should attach to.
+	//
+	// Deprecated: This field is deprecated since v1.44 and will be removed in a future release. The Networks field in
+	// TaskSpec should be used instead.
```
Comment (thaJeztah) [inline api/types/swarm/service.go:40]: FWIW; we won't be able to remove this from our code until we stop supporting API v1.43, but we can stop supporting it on newer API versions.


I guess for swarm services, there may be some additional challenges as the spec is stored in the cluster, or is that not an issue for this part of the code? 🤔 

/cc @dperny

**Review by thaJeztah — APPROVED** [review]
SGTM, but asked @dperny to also have a peek

**Comment by thaJeztah** [comment]
Looking at this one again; just in case I blame myself for not asking later on; do we have some handling in our code to detect if _both_ are set? Do you know what the current logic is if "old" is set or "new" is set (and what if _both_ are set, and they're not equal?)

**Comment by akerouanton** [comment]
Here's the relevant logic:

https://github.com/moby/moby/blob/dcf7287d647bcb515015e389df46ccf1e09855b7/daemon/cluster/networks.go#L304-L310

`populateNetworkID()` is called by `(*Cluster).CreateService()`, which is the daemon method called by `ServiceCreate` route handler.

Looking at the route handler, the daemon method and `populateNetworkID()`; I don't see any logic that would log or return a warning to the end-user.

**Review by cpuguy83 — DISMISSED** [review]

**Comment by akerouanton** [comment]
Conflict in version-history is fixed.

**Review by cpuguy83 — APPROVED** [review]

**Comment by thaJeztah** [comment]
Hm.. docker hub made a whoopsie with auth;

```
ERROR: failed to solve: tonistiigi/xx:1.2.1: failed to authorize: failed to fetch oauth token: Post "https://auth.docker.io/token": EOF
Error: buildx bake failed with: ERROR: failed to solve: tonistiigi/xx:1.2.1: failed to authorize: failed to fetch oauth token: Post "https://auth.docker.io/token": EOF
```

**Comment by akerouanton** [comment]
Yeah, GHA was grumpy when I pushed. Just restarted failed jobs -- hopefully everything should be green in a few minutes.

**Comment by akerouanton** [comment]
OK, all green now. Given that only wordings in Swagger doc changed since it was approved, let me bring this one in.

**Comment by thaJeztah** [comment]
Are we using these fields in the CLI? (if so, we'll probably need a re-vendor in docker/cli, and check if we're using these fields there / add `//nolint` comments where needed)

**Comment by akerouanton** [comment]
@thaJeztah Looks like we're using it. I'll add that to me todo-list.

**Pull request merged**